### DISCUSSION TITLE
Output as HTTP Server

### DISCUSSION THREAD

**Discussion by JelyMe** [opentrack/opentrack#1737] [Ideas]
An HTTP Server would be a very helpful output method. It would function like the UDP output option but instead work with HTTP GET requests. I would do it myself but I don't know enough C++ to read the code and jank together a solution.

**Comment by sthalik** [comment]
That's a lot of overhead, even with pipelining. Other output methods support emitting data at 250 Hz. You're better off using `api/plugin-support.hpp` inside your own main loop.

**Reply by JelyMe** [reply]
My goal is to access the data from a chrome extension, which obviously (for safety reasons) can't interact with the wider computer except through http. I can make an unoptimised python flask server to intercept the UDP packets and post to http but I thought it would be best to ask whether or not it was better to do it in C++

**Reply by sthalik** [reply]
Why not use websockets?

**Reply by JelyMe** [reply]
In a chrome extension? That's not exactly a safe thing so google prevents it. The only communication it has is http, which is soooo fun. I made a prototype in Python which just passes the UDP to HTTP and it works fine, no issues with performance which is a surprise coz python. I just wanted to know whether a native solution would work
